python-avd/pyavd/_eos_designs/structured_config/overlay/router_path_selection.py

```diff
@@ -45,9 +45,13 @@ def _cp_ipsec_profile_name(self: AvdStructuredConfigOverlay) -> str:
 
     @cached_property
     def _dp_ipsec_profile_name(self: AvdStructuredConfigOverlay) -> str:
-        """Returns the IPsec profile name to use for Data-Plane."""
-        # TODO: need to use CP one if 'wan_ipsec_profiles.data_plane' not present
-        return get(self._hostvars, "wan_ipsec_profiles.data_plane.profile_name", default="DP-PROFILE")
+        """Returns the IPsec profile name to use for Data-Plane.
+
+        If no data-plane config is present for IPsec, default to _cp_ipsec_profile_name
+        """
+        if (data_plane := get(self._hostvars, "wan_ipsec_profiles.data_plane")) is not None:
```

Comment:
It seems like this is the only place we are actually changing any behavior. And to me this seems more like a bugfix. The name here is only used for the `ha_path_group["ipsec_profile"]` field.

Reply:
yeah this is the TODO on the left